[JSC] Remove WasmLLInt

Constellation opened the pull request:
#### 83cc0906f0134ddeae14e327ab10342751a8eedf
<pre>
[JSC] Remove WasmLLInt
<a href="https://bugs.webkit.org/show_bug.cgi?id=297160">https://bugs.webkit.org/show_bug.cgi?id=297160</a>
<a href="https://rdar.apple.com/157895941">rdar://157895941</a>

Reviewed by Mark Lam.

Wasm IPInt is enabled for a while. It is time to remove Wams LLInt.
This is important since we start adding optimizations based on IPInt
assumption.

* JSTests/wasm/ipint-tests/perf.py:
* JSTests/wasm/stress/simd-regalloc-stress-2.js:
* JSTests/wasm/stress/simd-unreachable.js:
* JSTests/wasm/v8/adapter-frame.js:
* JSTests/wasm/v8/add-getters.js:
* JSTests/wasm/v8/anyfunc.js:
* JSTests/wasm/v8/array-copy-benchmark.js:
* JSTests/wasm/v8/array-init-from-segment.js:
* JSTests/wasm/v8/asm-wasm-copy.js:
* JSTests/wasm/v8/asm-wasm-deopt.js:
* JSTests/wasm/v8/asm-wasm-exception-in-tonumber.js:
* JSTests/wasm/v8/asm-wasm-expr.js:
* JSTests/wasm/v8/asm-wasm-f32.js:
* JSTests/wasm/v8/asm-wasm-f64.js:
* JSTests/wasm/v8/asm-wasm-heap.js:
* JSTests/wasm/v8/asm-wasm-i32.js:
* JSTests/wasm/v8/asm-wasm-imports.js:
* JSTests/wasm/v8/asm-wasm-literals.js:
* JSTests/wasm/v8/asm-wasm-math-intrinsic.js:
* JSTests/wasm/v8/asm-wasm-memory.js:
* JSTests/wasm/v8/asm-wasm-names.js:
* JSTests/wasm/v8/asm-wasm-stack.js:
* JSTests/wasm/v8/asm-wasm-stdlib.js:
* JSTests/wasm/v8/asm-wasm-switch.js:
* JSTests/wasm/v8/asm-wasm-u32.js:
* JSTests/wasm/v8/asm-wasm.js:
* JSTests/wasm/v8/asm-with-wasm-off.js:
* JSTests/wasm/v8/atomics-non-shared.js:
* JSTests/wasm/v8/atomics-stress.js:
* JSTests/wasm/v8/atomics.js:
* JSTests/wasm/v8/atomics64-stress.js:
* JSTests/wasm/v8/bigint-i64-to-imported-js-func.js:
* JSTests/wasm/v8/bigint-opt.js:
* JSTests/wasm/v8/bigint-rematerialize.js:
* JSTests/wasm/v8/bigint.js:
* JSTests/wasm/v8/bit-shift-right.js:
* JSTests/wasm/v8/bounds-check-64bit.js:
* JSTests/wasm/v8/bounds-check-turbofan.js:
* JSTests/wasm/v8/bulk-memory.js:
* JSTests/wasm/v8/call-ref.js:
* JSTests/wasm/v8/call_indirect.js:
* JSTests/wasm/v8/calls.js:
* JSTests/wasm/v8/code-space-overflow.js:
* JSTests/wasm/v8/committed-code-exhaustion.js:
* JSTests/wasm/v8/compare-exchange-stress.js:
* JSTests/wasm/v8/compare-exchange64-stress.js:
* JSTests/wasm/v8/compilation-hints-async-compilation.js:
* JSTests/wasm/v8/compilation-hints-decoder.js:
* JSTests/wasm/v8/compilation-hints-ignored.js:
* JSTests/wasm/v8/compilation-hints-lazy-validation.js:
* JSTests/wasm/v8/compilation-hints-streaming-compilation.js:
* JSTests/wasm/v8/compilation-hints-streaming-lazy-validation.js:
* JSTests/wasm/v8/compilation-hints-sync-compilation.js:
* JSTests/wasm/v8/compilation-limits-asm.js:
* JSTests/wasm/v8/compilation-limits.js:
* JSTests/wasm/v8/compiled-module-management.js:
* JSTests/wasm/v8/compiled-module-serialization.js:
* JSTests/wasm/v8/data-segments.js:
* JSTests/wasm/v8/disable-trap-handler.js:
* JSTests/wasm/v8/disallow-codegen.js:
* JSTests/wasm/v8/divrem-trap.js:
* JSTests/wasm/v8/element-segments-with-reftypes.js:
* JSTests/wasm/v8/empirical_max_memory.js:
* JSTests/wasm/v8/ensure-wasm-binaries-up-to-date.js:
* JSTests/wasm/v8/errors.js:
* JSTests/wasm/v8/export-global.js:
* JSTests/wasm/v8/export-identity.js:
* JSTests/wasm/v8/export-mutable-global.js:
* JSTests/wasm/v8/export-table.js:
* JSTests/wasm/v8/expose-wasm.js:
* JSTests/wasm/v8/extended-constants.js:
* JSTests/wasm/v8/externref-globals.js:
* JSTests/wasm/v8/externref-table.js:
* JSTests/wasm/v8/externref.js:
* JSTests/wasm/v8/ffi-error.js:
* JSTests/wasm/v8/ffi.js:
* JSTests/wasm/v8/float-constant-folding.js:
* JSTests/wasm/v8/function-names.js:
* JSTests/wasm/v8/function-prototype.js:
* JSTests/wasm/v8/futex.js:
* JSTests/wasm/v8/gc-buffer.js:
* JSTests/wasm/v8/gc-casts-from-any.js:
* JSTests/wasm/v8/gc-casts-invalid.js:
* JSTests/wasm/v8/gc-casts-subtypes.js:
* JSTests/wasm/v8/gc-experimental-string-conversions.js:
* JSTests/wasm/v8/gc-experiments.js:
* JSTests/wasm/v8/gc-frame.js:
* JSTests/wasm/v8/gc-js-interop-async-debugger.js:
* JSTests/wasm/v8/gc-js-interop-collections.js:
* JSTests/wasm/v8/gc-js-interop-export.mjs:
* JSTests/wasm/v8/gc-js-interop-global-constructors.js:
* JSTests/wasm/v8/gc-js-interop-import.mjs:
* JSTests/wasm/v8/gc-js-interop-numeric.js:
* JSTests/wasm/v8/gc-js-interop-objects.js:
* JSTests/wasm/v8/gc-js-interop-wasm.js:
* JSTests/wasm/v8/gc-js-interop.js:
* JSTests/wasm/v8/gc-memory.js:
* JSTests/wasm/v8/gc-nominal.js:
* JSTests/wasm/v8/gc-optimizations.js:
* JSTests/wasm/v8/gc-stress.js:
* JSTests/wasm/v8/gc-typecheck-reducer.js:
* JSTests/wasm/v8/gdbjit.js:
* JSTests/wasm/v8/generic-wrapper.js:
* JSTests/wasm/v8/globals-import-export-identity.js:
* JSTests/wasm/v8/globals.js:
* JSTests/wasm/v8/graceful_shutdown.js:
* JSTests/wasm/v8/graceful_shutdown_during_tierup.js:
* JSTests/wasm/v8/grow-huge-memory.js:
* JSTests/wasm/v8/grow-memory-detaching.js:
* JSTests/wasm/v8/grow-memory-in-branch.js:
* JSTests/wasm/v8/grow-memory-in-call.js:
* JSTests/wasm/v8/grow-memory-in-loop.js:
* JSTests/wasm/v8/grow-memory.js:
* JSTests/wasm/v8/grow-shared-memory.js:
* JSTests/wasm/v8/huge-memory.js:
* JSTests/wasm/v8/huge-typedarray.js:
* JSTests/wasm/v8/i31ref.js:
* JSTests/wasm/v8/import-function.js:
* JSTests/wasm/v8/import-memory.js:
* JSTests/wasm/v8/import-mutable-global.js:
* JSTests/wasm/v8/import-table.js:
* JSTests/wasm/v8/imported-function-types.js:
* JSTests/wasm/v8/incrementer.wasm:
* JSTests/wasm/v8/indirect-call-non-zero-table.js:
* JSTests/wasm/v8/indirect-calls.js:
* JSTests/wasm/v8/indirect-sig-mismatch.js:
* JSTests/wasm/v8/indirect-tables.js:
* JSTests/wasm/v8/inlining.js:
* JSTests/wasm/v8/instance-gc.js:
* JSTests/wasm/v8/instance-memory-gc-stress.js:
* JSTests/wasm/v8/instantiate-module-basic.js:
* JSTests/wasm/v8/instantiate-run-basic.js:
* JSTests/wasm/v8/js-api.js:
* JSTests/wasm/v8/large-offset.js:
* JSTests/wasm/v8/lazy-compilation.js:
* JSTests/wasm/v8/lazy-feedback-vector-allocation.js:
* JSTests/wasm/v8/liftoff-debug.js:
* JSTests/wasm/v8/liftoff-trap-handler.js:
* JSTests/wasm/v8/liftoff.js:
* JSTests/wasm/v8/load-immutable.js:
* JSTests/wasm/v8/log-code-after-post-message.js:
* JSTests/wasm/v8/loop-rotation.js:
* JSTests/wasm/v8/loop-unrolling.js:
* JSTests/wasm/v8/many-memories-no-trap-handler.js:
* JSTests/wasm/v8/many-memories.js:
* JSTests/wasm/v8/many-modules.js:
* JSTests/wasm/v8/many-parameters.js:
* JSTests/wasm/v8/max-module-size-flag.js:
* JSTests/wasm/v8/max-wasm-functions.js:
* JSTests/wasm/v8/memory-external-call.js:
* JSTests/wasm/v8/memory-instance-validation.js:
* JSTests/wasm/v8/memory-size.js:
* JSTests/wasm/v8/memory.js:
* JSTests/wasm/v8/memory64.js:
* JSTests/wasm/v8/memory_1gb_oob.js:
* JSTests/wasm/v8/memory_2gb_oob.js:
* JSTests/wasm/v8/memory_4gb_oob.js:
* JSTests/wasm/v8/module-memory.js:
* JSTests/wasm/v8/multi-table-element-section.js:
* JSTests/wasm/v8/multiple-code-spaces.js:
* JSTests/wasm/v8/mutable-globals.js:
* JSTests/wasm/v8/names.js:
* JSTests/wasm/v8/parallel_compilation.js:
* JSTests/wasm/v8/params.js:
* JSTests/wasm/v8/print-code.js:
* JSTests/wasm/v8/prototype.js:
* JSTests/wasm/v8/receiver.js:
* JSTests/wasm/v8/reference-globals-import.js:
* JSTests/wasm/v8/reference-globals.js:
* JSTests/wasm/v8/reference-table-js-interop.js:
* JSTests/wasm/v8/reference-tables.js:
* JSTests/wasm/v8/regress/regress-9017.js:
* JSTests/wasm/v8/resizablearraybuffer-growablesharedarraybuffer-wasm.js:
* JSTests/wasm/v8/return-calls.js:
* JSTests/wasm/v8/runtime-type-canonicalization.js:
* JSTests/wasm/v8/serialization-with-compilation-hints.js:
* JSTests/wasm/v8/serialize-lazy-module.js:
* JSTests/wasm/v8/shared-arraybuffer-worker-simple-gc.js:
* JSTests/wasm/v8/shared-memory-gc-stress.js:
* JSTests/wasm/v8/shared-memory-worker-explicit-gc-stress.js:
* JSTests/wasm/v8/shared-memory-worker-gc-stress.js:
* JSTests/wasm/v8/shared-memory-worker-gc.js:
* JSTests/wasm/v8/shared-memory-worker-simple-gc.js:
* JSTests/wasm/v8/shared-memory-worker-stress.js:
* JSTests/wasm/v8/shared-memory.js:
* JSTests/wasm/v8/single-threaded-compilation.js:
* JSTests/wasm/v8/speculative-inlining.js:
* JSTests/wasm/v8/stack-switching-export.js:
* JSTests/wasm/v8/stack-switching.js:
* JSTests/wasm/v8/stack.js:
* JSTests/wasm/v8/stackwalk.js:
* JSTests/wasm/v8/start-function.js:
* JSTests/wasm/v8/streaming-api.js:
* JSTests/wasm/v8/streaming-compile.js:
* JSTests/wasm/v8/streaming-error-position.js:
* JSTests/wasm/v8/streaming-trap-location.js:
* JSTests/wasm/v8/stringrefs-exec-gc.js:
* JSTests/wasm/v8/stringrefs-exec.js:
* JSTests/wasm/v8/stringrefs-invalid.js:
* JSTests/wasm/v8/stringrefs-js.js:
* JSTests/wasm/v8/stringrefs-regressions.js:
* JSTests/wasm/v8/stringrefs-valid.js:
* JSTests/wasm/v8/table-access.js:
* JSTests/wasm/v8/table-copy-externref.js:
* JSTests/wasm/v8/table-copy.js:
* JSTests/wasm/v8/table-fill.js:
* JSTests/wasm/v8/table-get.js:
* JSTests/wasm/v8/table-grow-from-wasm.js:
* JSTests/wasm/v8/table-grow.js:
* JSTests/wasm/v8/table-limits.js:
* JSTests/wasm/v8/tagged-stack-parameters.js:
* JSTests/wasm/v8/test-partial-serialization.js:
* JSTests/wasm/v8/test-serialization-with-lazy-compilation.js:
* JSTests/wasm/v8/test-wasm-module-builder.js:
* JSTests/wasm/v8/tier-down-to-liftoff.js:
* JSTests/wasm/v8/tier-up-testing-flag.js:
* JSTests/wasm/v8/type-based-optimizations.js:
* JSTests/wasm/v8/type-reflection-with-externref.js:
* JSTests/wasm/v8/type-reflection-with-mv.js:
* JSTests/wasm/v8/type-reflection.js:
* JSTests/wasm/v8/typed-funcref.js:
* JSTests/wasm/v8/unicode-validation.js:
* JSTests/wasm/v8/unicode.js:
* JSTests/wasm/v8/unreachable-validation.js:
* JSTests/wasm/v8/unreachable.js:
* JSTests/wasm/v8/user-properties-constructed.js:
* JSTests/wasm/v8/user-properties-exported.js:
* JSTests/wasm/v8/user-properties-module.js:
* JSTests/wasm/v8/user-properties-reexport.js:
* JSTests/wasm/v8/verify-module-basic-errors.js:
* JSTests/wasm/v8/wasm-api-overloading.js:
* JSTests/wasm/v8/wasm-default.js:
* JSTests/wasm/v8/wasm-dynamic-tiering.js:
* JSTests/wasm/v8/wasm-gc-externalize-internalize.js:
* JSTests/wasm/v8/wasm-gc-js-ref.js:
* JSTests/wasm/v8/wasm-gc-js-roundtrip.js:
* JSTests/wasm/v8/wasm-invalid-local.js:
* JSTests/wasm/v8/wasm-math-intrinsic.js:
* JSTests/wasm/v8/wasm-object-api.js:
* JSTests/wasm/v8/worker-memory.js:
* JSTests/wasm/v8/worker-module.js:
* JSTests/wasm/v8/worker-running-empty-loop-interruptible.js:
* Source/JavaScriptCore/CMakeLists.txt:
* Source/JavaScriptCore/DerivedSources.make:
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/JavaScriptCore/SaferCPPExpectations/MemoryUnsafeCastCheckerExpectations:
* Source/JavaScriptCore/SaferCPPExpectations/NoUncountedMemberCheckerExpectations:
* Source/JavaScriptCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/JavaScriptCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/JavaScriptCore/Sources.txt:
* Source/JavaScriptCore/assembler/JITOperationList.cpp:
(JSC::llintOperations):
* Source/JavaScriptCore/b3/B3Procedure.h:
(JSC::B3::Procedure::usesSIMD const):
* Source/JavaScriptCore/bytecode/BytecodeBasicBlock.h:
* Source/JavaScriptCore/bytecode/BytecodeDumper.cpp:
(JSC::Wasm::BytecodeDumper::dumpBlock): Deleted.
(JSC::Wasm::BytecodeDumper::dumpConstants): Deleted.
(JSC::Wasm::BytecodeDumper::dumpExceptionHandlers): Deleted.
(JSC::Wasm::BytecodeDumper::constantName const): Deleted.
(JSC::Wasm::BytecodeDumper::formatConstant const): Deleted.
(JSC::Wasm::BytecodeDumper::registerName const): Deleted.
(JSC::Wasm::BytecodeDumper::outOfLineJumpOffset const): Deleted.
* Source/JavaScriptCore/bytecode/BytecodeDumper.h:
* Source/JavaScriptCore/bytecode/BytecodeList.rb:
* Source/JavaScriptCore/bytecode/Instruction.h:
* Source/JavaScriptCore/bytecode/InstructionStream.h:
* Source/JavaScriptCore/bytecode/Opcode.cpp:
* Source/JavaScriptCore/bytecode/Opcode.h:
* Source/JavaScriptCore/generator/DSL.rb:
* Source/JavaScriptCore/generator/Options.rb:
* Source/JavaScriptCore/generator/Wasm.rb:
* Source/JavaScriptCore/interpreter/Interpreter.cpp:
(JSC::CatchInfo::CatchInfo):
* Source/JavaScriptCore/interpreter/Interpreter.h:
* Source/JavaScriptCore/jit/JITExceptions.cpp:
(JSC::genericUnwind):
* Source/JavaScriptCore/llint/InPlaceInterpreter.asm:
* Source/JavaScriptCore/llint/InPlaceInterpreter64.asm:
* Source/JavaScriptCore/llint/LLIntData.cpp:
(JSC::LLInt::neuterOpcodeMaps):
(JSC::LLInt::initialize):
* Source/JavaScriptCore/llint/LLIntData.h:
(JSC::LLInt::wasmExceptionInstructions): Deleted.
* Source/JavaScriptCore/llint/LLIntExceptions.cpp:
(JSC::LLInt::wasmReturnToThrow): Deleted.
(JSC::LLInt::handleWasmCatch): Deleted.
(JSC::LLInt::handleWasmCatchAll): Deleted.
(JSC::LLInt::handleWasmTryTable): Deleted.
* Source/JavaScriptCore/llint/LLIntExceptions.h:
* Source/JavaScriptCore/llint/LLIntOffsetsExtractor.cpp:
* Source/JavaScriptCore/llint/LLIntThunks.cpp:
(JSC::LLInt::wasmFunctionEntryThunk): Deleted.
(JSC::LLInt::wasmFunctionEntryThunkSIMD): Deleted.
(JSC::LLInt::handleWasmCatchThunk): Deleted.
(JSC::LLInt::handleWasmCatchAllThunk): Deleted.
(JSC::LLInt::handleWasmTryTableThunk): Deleted.
* Source/JavaScriptCore/llint/LLIntThunks.h:
* Source/JavaScriptCore/llint/LowLevelInterpreter.asm:
* Source/JavaScriptCore/llint/LowLevelInterpreter32_64.asm:
* Source/JavaScriptCore/llint/LowLevelInterpreter64.asm:
* Source/JavaScriptCore/llint/WebAssembly.asm:
* Source/JavaScriptCore/llint/WebAssembly32_64.asm: Removed.
* Source/JavaScriptCore/llint/WebAssembly64.asm: Removed.
* Source/JavaScriptCore/runtime/BufferMemoryHandle.cpp:
(JSC::BufferMemoryManager::isInGrowableOrFastMemory):
* Source/JavaScriptCore/runtime/Gate.h:
* Source/JavaScriptCore/runtime/JSCConfig.h:
* Source/JavaScriptCore/runtime/Options.cpp:
(JSC::Options::isAvailable):
(JSC::disableAllWasmOptions):
(JSC::Options::notifyOptionsChanged):
(JSC::Options::assertOptionsAreCoherent):
* Source/JavaScriptCore/runtime/OptionsList.h:
* Source/JavaScriptCore/runtime/SamplingProfiler.cpp:
(JSC::tierName):
* Source/JavaScriptCore/tools/JSDollarVM.cpp:
(JSC::JSC_DEFINE_HOST_FUNCTION):
* Source/JavaScriptCore/wasm/WasmBBQJIT.cpp:
(JSC::Wasm::BBQJITImpl::BBQJIT::addEndToUnreachable):
* Source/JavaScriptCore/wasm/WasmBBQPlan.cpp:
(JSC::Wasm::BBQPlan::work):
* Source/JavaScriptCore/wasm/WasmCallee.cpp:
(JSC::Wasm::Callee::runWithDowncast):
(JSC::Wasm::LLIntCallee::LLIntCallee): Deleted.
(JSC::Wasm::LLIntCallee::setEntrypoint): Deleted.
(JSC::Wasm::LLIntCallee::calleeSaveRegistersImpl): Deleted.
(JSC::Wasm::LLIntCallee::outOfLineJumpOffset): Deleted.
(JSC::Wasm::LLIntCallee::outOfLineJumpTarget): Deleted.
* Source/JavaScriptCore/wasm/WasmCallee.h:
* Source/JavaScriptCore/wasm/WasmCalleeGroup.cpp:
(JSC::Wasm::CalleeGroup::CalleeGroup):
(JSC::Wasm::CalleeGroup::releaseBBQCallee):
(JSC::Wasm::CalleeGroup::calleeIsReferenced const):
(JSC::Wasm::CalleeGroup::createFromLLInt): Deleted.
* Source/JavaScriptCore/wasm/WasmCalleeGroup.h:
* Source/JavaScriptCore/wasm/WasmCompilationMode.h:
(JSC::Wasm::isAnyInterpreter):
(JSC::Wasm::isAnyBBQ):
(JSC::Wasm::isAnyOMG):
(JSC::Wasm::isAnyWasmToJS):
* Source/JavaScriptCore/wasm/WasmEntryPlan.cpp:
(JSC::Wasm::EntryPlan::generateWasmToWasmStubs):
* Source/JavaScriptCore/wasm/WasmFaultSignalHandler.cpp:
(JSC::Wasm::trapHandler):
* Source/JavaScriptCore/wasm/WasmFunctionCodeBlockGenerator.cpp: Removed.
* Source/JavaScriptCore/wasm/WasmFunctionCodeBlockGenerator.h: Removed.
* Source/JavaScriptCore/wasm/WasmGeneratorTraits.h: Removed.
* Source/JavaScriptCore/wasm/WasmGlobal.h:
* Source/JavaScriptCore/wasm/WasmIPIntGenerator.cpp:
* Source/JavaScriptCore/wasm/WasmIPIntPlan.h:
* Source/JavaScriptCore/wasm/WasmIPIntSlowPaths.cpp:
(JSC::IPInt::wasm_log_crash):
(JSC::IPInt::slow_path_wasm_throw_exception):
(JSC::IPInt::slow_path_wasm_popcount):
(JSC::IPInt::slow_path_wasm_popcountll):
* Source/JavaScriptCore/wasm/WasmIPIntSlowPaths.h:
* Source/JavaScriptCore/wasm/WasmLLIntBuiltin.h: Removed.
* Source/JavaScriptCore/wasm/WasmLLIntGenerator.cpp: Removed.
* Source/JavaScriptCore/wasm/WasmLLIntGenerator.h: Removed.
* Source/JavaScriptCore/wasm/WasmLLIntPlan.cpp: Removed.
* Source/JavaScriptCore/wasm/WasmLLIntPlan.h: Removed.
* Source/JavaScriptCore/wasm/WasmLLIntTierUpCounter.cpp:
* Source/JavaScriptCore/wasm/WasmLLIntTierUpCounter.h:
* Source/JavaScriptCore/wasm/WasmModule.cpp:
(JSC::Wasm::Module::Module):
(JSC::Wasm::makeValidationCallback):
(JSC::Wasm::Module::validateSync):
(JSC::Wasm::Module::validateAsync):
(JSC::Wasm::Module::getOrCreateCalleeGroup):
* Source/JavaScriptCore/wasm/WasmModule.h:
* Source/JavaScriptCore/wasm/WasmModuleInformation.h:
* Source/JavaScriptCore/wasm/WasmOMGIRGenerator.cpp:
(JSC::Wasm::OMGIRGenerator::addCall):
* Source/JavaScriptCore/wasm/WasmOMGPlan.cpp:
(JSC::Wasm::OMGPlan::work):
* Source/JavaScriptCore/wasm/WasmOperations.cpp:
(JSC::Wasm::JSC_DEFINE_NOEXCEPT_JIT_OPERATION):
* Source/JavaScriptCore/wasm/WasmOperations.h:
* Source/JavaScriptCore/wasm/WasmOperationsInlines.h:
* Source/JavaScriptCore/wasm/WasmSlowPaths.cpp: Removed.
* Source/JavaScriptCore/wasm/WasmSlowPaths.h: Removed.
* Source/JavaScriptCore/wasm/WasmStreamingCompiler.cpp:
(JSC::Wasm::StreamingCompiler::didReceiveFunctionData):
(JSC::Wasm::StreamingCompiler::didFinishParsing):
(JSC::Wasm::StreamingCompiler::didComplete):
* Source/JavaScriptCore/wasm/WasmTypeDefinition.cpp:
(JSC::Wasm::TypeInformation::signatureForLLIntBuiltin): Deleted.
* Source/JavaScriptCore/wasm/WasmTypeDefinition.h:
* Source/JavaScriptCore/wasm/WasmWorklist.cpp:
* Source/JavaScriptCore/wasm/js/JSWebAssemblyInstance.cpp:
(JSC::JSWebAssemblyInstance::finalizeCreation):
* Source/JavaScriptCore/wasm/js/JSWebAssemblyStruct.cpp:
(JSC::JSWebAssemblyStruct::get const):
* Source/JavaScriptCore/wasm/js/WebAssemblyFunction.h:
* Tools/Scripts/run-jsc-stress-tests:

Canonical link: <a href="https://commits.webkit.org/298474@main">https://commits.webkit.org/298474@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/834a9d91d23d125e428f7507063ff33ff9f19039

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/115620 "3 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/35283 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/25806 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/121667 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/66130 "Built successfully") | [❌ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/8c0bbd89-9138-4cb7-8f43-307dd857deda) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/117509 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/35964 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/43866 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/87830 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/42457 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/cefc82a3-e190-4372-83f4-a2906a8ac516) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/118568 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/28688 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/103764 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/68229 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/27825 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/21880 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/65346 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/107742 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/98064 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/22001 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/124816 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/114146 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/42515 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/31872 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/96584 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/42882 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/99954 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96372 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/41627 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/19483 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/38383 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18480 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/42406 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/47978 "Built successfully") | [⏳ 🛠 jsc-armv7 ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/41879 "Built successfully") | | [⏳ 🧪 jsc-armv7-tests ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/45210 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/43587 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

sosukesuzuki commented on Source/JavaScriptCore/runtime/OptionsList.h:
```diff
@@ -557,12 +557,8 @@ bool hasCapacityToUseLargeGigacage();
     v(Bool, verboseBBQJITAllocation, false, Normal, "Logs extra information about register allocation during BBQ JIT"_s) \
     v(Bool, verboseBBQJITInstructions, false, Normal, "Logs instruction information during BBQ JIT"_s) \
     v(Bool, disableBBQConsts, false, Normal, "Wasm <type>.const instructions in BBQ JIT won't lower to a const BBQ::Value"_s) \
-    v(Bool, useWasmLLInt, true, Normal, nullptr) \
```
Won't you remove the `useWasmIPInt` option? I'm curious about what would happen if we run Wasm with `useWasmIPInt=false` after this change.

Constellation replied:
Start using BBQ immediately.

sosukesuzuki replied:
Ah I got it, thank you

MenloDorian reviewed (approved):
rs=me if EWS bots are green.

Constellation committed:
[JSC] Remove WasmLLInt
https://bugs.webkit.org/show_bug.cgi?id=297160
rdar://157895941

Reviewed by Mark Lam.

Wasm IPInt is enabled for a while. It is time to remove Wams LLInt.
This is important since we start adding optimizations based on IPInt
assumption.

* JSTests/wasm/ipint-tests/perf.py:
* JSTests/wasm/stress/simd-regalloc-stress-2.js:
* JSTests/wasm/stress/simd-unreachable.js:
* JSTests/wasm/v8/adapter-frame.js:
* JSTests/wasm/v8/add-getters.js:
* JSTests/wasm/v8/anyfunc.js:
* JSTests/wasm/v8/array-copy-benchmark.js:
* JSTests/wasm/v8/array-init-from-segment.js:
* JSTests/wasm/v8/asm-wasm-copy.js:
* JSTests/wasm/v8/asm-wasm-deopt.js:
* JSTests/wasm/v8/asm-wasm-exception-in-tonumber.js:
* JSTests/wasm/v8/asm-wasm-expr.js:
* JSTests/wasm/v8/asm-wasm-f32.js:
* JSTests/wasm/v8/asm-wasm-f64.js:
* JSTests/wasm/v8/asm-wasm-heap.js:
* JSTests/wasm/v8/asm-wasm-i32.js:
* JSTests/wasm/v8/asm-wasm-imports.js:
* JSTests/wasm/v8/asm-wasm-literals.js:
* JSTests/wasm/v8/asm-wasm-math-intrinsic.js:
* JSTests/wasm/v8/asm-wasm-memory.js:
* JSTests/wasm/v8/asm-wasm-names.js:
* JSTests/wasm/v8/asm-wasm-stack.js:
* JSTests/wasm/v8/asm-wasm-stdlib.js:
* JSTests/wasm/v8/asm-wasm-switch.js:
* JSTests/wasm/v8/asm-wasm-u32.js:
* JSTests/wasm/v8/asm-wasm.js:
* JSTests/wasm/v8/asm-with-wasm-off.js:
* JSTests/wasm/v8/atomics-non-shared.js:
* JSTests/wasm/v8/atomics-stress.js:
* JSTests/wasm/v8/atomics.js:
* JSTests/wasm/v8/atomics64-stress.js:
* JSTests/wasm/v8/bigint-i64-to-imported-js-func.js:
* JSTests/wasm/v8/bigint-opt.js:
* JSTests/wasm/v8/bigint-rematerialize.js:
* JSTests/wasm/v8/bigint.js:
* JSTests/wasm/v8/bit-shift-right.js:
* JSTests/wasm/v8/bounds-check-64bit.js:
* JSTests/wasm/v8/bounds-check-turbofan.js:
* JSTests/wasm/v8/bulk-memory.js:
* JSTests/wasm/v8/call-ref.js:
* JSTests/wasm/v8/call_indirect.js:
* JSTests/wasm/v8/calls.js:
* JSTests/wasm/v8/code-space-overflow.js:
* JSTests/wasm/v8/committed-code-exhaustion.js:
* JSTests/wasm/v8/compare-exchange-stress.js:
* JSTests/wasm/v8/compare-exchange64-stress.js:
* JSTests/wasm/v8/compilation-hints-async-compilation.js:
* JSTests/wasm/v8/compilation-hints-decoder.js:
* JSTests/wasm/v8/compilation-hints-ignored.js:
* JSTests/wasm/v8/compilation-hints-lazy-validation.js:
* JSTests/wasm/v8/compilation-hints-streaming-compilation.js:
* JSTests/wasm/v8/compilation-hints-streaming-lazy-validation.js:
* JSTests/wasm/v8/compilation-hints-sync-compilation.js:
* JSTests/wasm/v8/compilation-limits-asm.js:
* JSTests/wasm/v8/compilation-limits.js:
* JSTests/wasm/v8/compiled-module-management.js:
* JSTests/wasm/v8/compiled-module-serialization.js:
* JSTests/wasm/v8/data-segments.js:
* JSTests/wasm/v8/disable-trap-handler.js:
* JSTests/wasm/v8/disallow-codegen.js:
* JSTests/wasm/v8/divrem-trap.js:
* JSTests/wasm/v8/element-segments-with-reftypes.js:
* JSTests/wasm/v8/empirical_max_memory.js:
* JSTests/wasm/v8/ensure-wasm-binaries-up-to-date.js:
* JSTests/wasm/v8/errors.js:
* JSTests/wasm/v8/export-global.js:
* JSTests/wasm/v8/export-identity.js:
* JSTests/wasm/v8/export-mutable-global.js:
* JSTests/wasm/v8/export-table.js:
* JSTests/wasm/v8/expose-wasm.js:
* JSTests/wasm/v8/extended-constants.js:
* JSTests/wasm/v8/externref-globals.js:
* JSTests/wasm/v8/externref-table.js:
* JSTests/wasm/v8/externref.js:
* JSTests/wasm/v8/ffi-error.js:
* JSTests/wasm/v8/ffi.js:
* JSTests/wasm/v8/float-constant-folding.js:
* JSTests/wasm/v8/function-names.js:
* JSTests/wasm/v8/function-prototype.js:
* JSTests/wasm/v8/futex.js:
* JSTests/wasm/v8/gc-buffer.js:
* JSTests/wasm/v8/gc-casts-from-any.js:
* JSTests/wasm/v8/gc-casts-invalid.js:
* JSTests/wasm/v8/gc-casts-subtypes.js:
* JSTests/wasm/v8/gc-experimental-string-conversions.js:
* JSTests/wasm/v8/gc-experiments.js:
* JSTests/wasm/v8/gc-frame.js:
* JSTests/wasm/v8/gc-js-interop-async-debugger.js:
* JSTests/wasm/v8/gc-js-interop-collections.js:
* JSTests/wasm/v8/gc-js-interop-export.mjs:
* JSTests/wasm/v8/gc-js-interop-global-constructors.js:
* JSTests/wasm/v8/gc-js-interop-import.mjs:
* JSTests/wasm/v8/gc-js-interop-numeric.js:
* JSTests/wasm/v8/gc-js-interop-objects.js:
* JSTests/wasm/v8/gc-js-interop-wasm.js:
* JSTests/wasm/v8/gc-js-interop.js:
* JSTests/wasm/v8/gc-memory.js:
* JSTests/wasm/v8/gc-nominal.js:
* JSTests/wasm/v8/gc-optimizations.js:
* JSTests/wasm/v8/gc-stress.js:
* JSTests/wasm/v8/gc-typecheck-reducer.js:
* JSTests/wasm/v8/gdbjit.js:
* JSTests/wasm/v8/generic-wrapper.js:
* JSTests/wasm/v8/globals-import-export-identity.js:
* JSTests/wasm/v8/globals.js:
* JSTests/wasm/v8/graceful_shutdown.js:
* JSTests/wasm/v8/graceful_shutdown_during_tierup.js:
* JSTests/wasm/v8/grow-huge-memory.js:
* JSTests/wasm/v8/grow-memory-detaching.js:
* JSTests/wasm/v8/grow-memory-in-branch.js:
* JSTests/wasm/v8/grow-memory-in-call.js:
* JSTests/wasm/v8/grow-memory-in-loop.js:
* JSTests/wasm/v8/grow-memory.js:
* JSTests/wasm/v8/grow-shared-memory.js:
* JSTests/wasm/v8/huge-memory.js:
* JSTests/wasm/v8/huge-typedarray.js:
* JSTests/wasm/v8/i31ref.js:
* JSTests/wasm/v8/import-function.js:
* JSTests/wasm/v8/import-memory.js:
* JSTests/wasm/v8/import-mutable-global.js:
* JSTests/wasm/v8/import-table.js:
* JSTests/wasm/v8/imported-function-types.js:
* JSTests/wasm/v8/incrementer.wasm:
* JSTests/wasm/v8/indirect-call-non-zero-table.js:
* JSTests/wasm/v8/indirect-calls.js:
* JSTests/wasm/v8/indirect-sig-mismatch.js:
* JSTests/wasm/v8/indirect-tables.js:
* JSTests/wasm/v8/inlining.js:
* JSTests/wasm/v8/instance-gc.js:
* JSTests/wasm/v8/instance-memory-gc-stress.js:
* JSTests/wasm/v8/instantiate-module-basic.js:
* JSTests/wasm/v8/instantiate-run-basic.js:
* JSTests/wasm/v8/js-api.js:
* JSTests/wasm/v8/large-offset.js:
* JSTests/wasm/v8/lazy-compilation.js:
* JSTests/wasm/v8/lazy-feedback-vector-allocation.js:
* JSTests/wasm/v8/liftoff-debug.js:
* JSTests/wasm/v8/liftoff-trap-handler.js:
* JSTests/wasm/v8/liftoff.js:
* JSTests/wasm/v8/load-immutable.js:
* JSTests/wasm/v8/log-code-after-post-message.js:
* JSTests/wasm/v8/loop-rotation.js:
* JSTests/wasm/v8/loop-unrolling.js:
* JSTests/wasm/v8/many-memories-no-trap-handler.js:
* JSTests/wasm/v8/many-memories.js:
* JSTests/wasm/v8/many-modules.js:
* JSTests/wasm/v8/many-parameters.js:
* JSTests/wasm/v8/max-module-size-flag.js:
* JSTests/wasm/v8/max-wasm-functions.js:
* JSTests/wasm/v8/memory-external-call.js:
* JSTests/wasm/v8/memory-instance-validation.js:
* JSTests/wasm/v8/memory-size.js:
* JSTests/wasm/v8/memory.js:
* JSTests/wasm/v8/memory64.js:
* JSTests/wasm/v8/memory_1gb_oob.js:
* JSTests/wasm/v8/memory_2gb_oob.js:
* JSTests/wasm/v8/memory_4gb_oob.js:
* JSTests/wasm/v8/module-memory.js:
* JSTests/wasm/v8/multi-table-element-section.js:
* JSTests/wasm/v8/multiple-code-spaces.js:
* JSTests/wasm/v8/mutable-globals.js:
* JSTests/wasm/v8/names.js:
* JSTests/wasm/v8/parallel_compilation.js:
* JSTests/wasm/v8/params.js:
* JSTests/wasm/v8/print-code.js:
* JSTests/wasm/v8/prototype.js:
* JSTests/wasm/v8/receiver.js:
* JSTests/wasm/v8/reference-globals-import.js:
* JSTests/wasm/v8/reference-globals.js:
* JSTests/wasm/v8/reference-table-js-interop.js:
* JSTests/wasm/v8/reference-tables.js:
* JSTests/wasm/v8/regress/regress-9017.js:
* JSTests/wasm/v8/resizablearraybuffer-growablesharedarraybuffer-wasm.js:
* JSTests/wasm/v8/return-calls.js:
* JSTests/wasm/v8/runtime-type-canonicalization.js:
* JSTests/wasm/v8/serialization-with-compilation-hints.js:
* JSTests/wasm/v8/serialize-lazy-module.js:
* JSTests/wasm/v8/shared-arraybuffer-worker-simple-gc.js:
* JSTests/wasm/v8/shared-memory-gc-stress.js:
* JSTests/wasm/v8/shared-memory-worker-explicit-gc-stress.js:
* JSTests/wasm/v8/shared-memory-worker-gc-stress.js:
* JSTests/wasm/v8/shared-memory-worker-gc.js:
* JSTests/wasm/v8/shared-memory-worker-simple-gc.js:
* JSTests/wasm/v8/shared-memory-worker-stress.js:
* JSTests/wasm/v8/shared-memory.js:
* JSTests/wasm/v8/single-threaded-compilation.js:
* JSTests/wasm/v8/speculative-inlining.js:
* JSTests/wasm/v8/stack-switching-export.js:
* JSTests/wasm/v8/stack-switching.js:
* JSTests/wasm/v8/stack.js:
* JSTests/wasm/v8/stackwalk.js:
* JSTests/wasm/v8/start-function.js:
* JSTests/wasm/v8/streaming-api.js:
* JSTests/wasm/v8/streaming-compile.js:
* JSTests/wasm/v8/streaming-error-position.js:
* JSTests/wasm/v8/streaming-trap-location.js:
* JSTests/wasm/v8/stringrefs-exec-gc.js:
* JSTests/wasm/v8/stringrefs-exec.js:
* JSTests/wasm/v8/stringrefs-invalid.js:
* JSTests/wasm/v8/stringrefs-js.js:
* JSTests/wasm/v8/stringrefs-regressions.js:
* JSTests/wasm/v8/stringrefs-valid.js:
* JSTests/wasm/v8/table-access.js:
* JSTests/wasm/v8/table-copy-externref.js:
* JSTests/wasm/v8/table-copy.js:
* JSTests/wasm/v8/table-fill.js:
* JSTests/wasm/v8/table-get.js:
* JSTests/wasm/v8/table-grow-from-wasm.js:
* JSTests/wasm/v8/table-grow.js:
* JSTests/wasm/v8/table-limits.js:
* JSTests/wasm/v8/tagged-stack-parameters.js:
* JSTests/wasm/v8/test-partial-serialization.js:
* JSTests/wasm/v8/test-serialization-with-lazy-compilation.js:
* JSTests/wasm/v8/test-wasm-module-builder.js:
* JSTests/wasm/v8/tier-down-to-liftoff.js:
* JSTests/wasm/v8/tier-up-testing-flag.js:
* JSTests/wasm/v8/type-based-optimizations.js:
* JSTests/wasm/v8/type-reflection-with-externref.js:
* JSTests/wasm/v8/type-reflection-with-mv.js:
* JSTests/wasm/v8/type-reflection.js:
* JSTests/wasm/v8/typed-funcref.js:
* JSTests/wasm/v8/unicode-validation.js:
* JSTests/wasm/v8/unicode.js:
* JSTests/wasm/v8/unreachable-validation.js:
* JSTests/wasm/v8/unreachable.js:
* JSTests/wasm/v8/user-properties-constructed.js:
* JSTests/wasm/v8/user-properties-exported.js:
* JSTests/wasm/v8/user-properties-module.js:
* JSTests/wasm/v8/user-properties-reexport.js:
* JSTests/wasm/v8/verify-module-basic-errors.js:
* JSTests/wasm/v8/wasm-api-overloading.js:
* JSTests/wasm/v8/wasm-default.js:
* JSTests/wasm/v8/wasm-dynamic-tiering.js:
* JSTests/wasm/v8/wasm-gc-externalize-internalize.js:
* JSTests/wasm/v8/wasm-gc-js-ref.js:
* JSTests/wasm/v8/wasm-gc-js-roundtrip.js:
* JSTests/wasm/v8/wasm-invalid-local.js:
* JSTests/wasm/v8/wasm-math-intrinsic.js:
* JSTests/wasm/v8/wasm-object-api.js:
* JSTests/wasm/v8/worker-memory.js:
* JSTests/wasm/v8/worker-module.js:
* JSTests/wasm/v8/worker-running-empty-loop-interruptible.js:
* Source/JavaScriptCore/CMakeLists.txt:
* Source/JavaScriptCore/DerivedSources.make:
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/JavaScriptCore/SaferCPPExpectations/MemoryUnsafeCastCheckerExpectations:
* Source/JavaScriptCore/SaferCPPExpectations/NoUncountedMemberCheckerExpectations:
* Source/JavaScriptCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/JavaScriptCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/JavaScriptCore/Sources.txt:
* Source/JavaScriptCore/assembler/JITOperationList.cpp:
(JSC::llintOperations):
* Source/JavaScriptCore/b3/B3Procedure.h:
(JSC::B3::Procedure::usesSIMD const):
* Source/JavaScriptCore/bytecode/BytecodeBasicBlock.h:
* Source/JavaScriptCore/bytecode/BytecodeDumper.cpp:
(JSC::Wasm::BytecodeDumper::dumpBlock): Deleted.
(JSC::Wasm::BytecodeDumper::dumpConstants): Deleted.
(JSC::Wasm::BytecodeDumper::dumpExceptionHandlers): Deleted.
(JSC::Wasm::BytecodeDumper::constantName const): Deleted.
(JSC::Wasm::BytecodeDumper::formatConstant const): Deleted.
(JSC::Wasm::BytecodeDumper::registerName const): Deleted.
(JSC::Wasm::BytecodeDumper::outOfLineJumpOffset const): Deleted.
* Source/JavaScriptCore/bytecode/BytecodeDumper.h:
* Source/JavaScriptCore/bytecode/BytecodeList.rb:
* Source/JavaScriptCore/bytecode/Instruction.h:
* Source/JavaScriptCore/bytecode/InstructionStream.h:
* Source/JavaScriptCore/bytecode/Opcode.cpp:
* Source/JavaScriptCore/bytecode/Opcode.h:
* Source/JavaScriptCore/generator/DSL.rb:
* Source/JavaScriptCore/generator/Options.rb:
* Source/JavaScriptCore/generator/Wasm.rb:
* Source/JavaScriptCore/interpreter/Interpreter.cpp:
(JSC::CatchInfo::CatchInfo):
* Source/JavaScriptCore/interpreter/Interpreter.h:
* Source/JavaScriptCore/jit/JITExceptions.cpp:
(JSC::genericUnwind):
* Source/JavaScriptCore/llint/InPlaceInterpreter.asm:
* Source/JavaScriptCore/llint/InPlaceInterpreter64.asm:
* Source/JavaScriptCore/llint/LLIntData.cpp:
(JSC::LLInt::neuterOpcodeMaps):
(JSC::LLInt::initialize):
* Source/JavaScriptCore/llint/LLIntData.h:
(JSC::LLInt::wasmExceptionInstructions): Deleted.
* Source/JavaScriptCore/llint/LLIntExceptions.cpp:
(JSC::LLInt::wasmReturnToThrow): Deleted.
(JSC::LLInt::handleWasmCatch): Deleted.
(JSC::LLInt::handleWasmCatchAll): Deleted.
(JSC::LLInt::handleWasmTryTable): Deleted.
* Source/JavaScriptCore/llint/LLIntExceptions.h:
* Source/JavaScriptCore/llint/LLIntOffsetsExtractor.cpp:
* Source/JavaScriptCore/llint/LLIntThunks.cpp:
(JSC::LLInt::wasmFunctionEntryThunk): Deleted.
(JSC::LLInt::wasmFunctionEntryThunkSIMD): Deleted.
(JSC::LLInt::handleWasmCatchThunk): Deleted.
(JSC::LLInt::handleWasmCatchAllThunk): Deleted.
(JSC::LLInt::handleWasmTryTableThunk): Deleted.
* Source/JavaScriptCore/llint/LLIntThunks.h:
* Source/JavaScriptCore/llint/LowLevelInterpreter.asm:
* Source/JavaScriptCore/llint/LowLevelInterpreter32_64.asm:
* Source/JavaScriptCore/llint/LowLevelInterpreter64.asm:
* Source/JavaScriptCore/llint/WebAssembly.asm:
* Source/JavaScriptCore/llint/WebAssembly32_64.asm: Removed.
* Source/JavaScriptCore/llint/WebAssembly64.asm: Removed.
* Source/JavaScriptCore/runtime/BufferMemoryHandle.cpp:
(JSC::BufferMemoryManager::isInGrowableOrFastMemory):
* Source/JavaScriptCore/runtime/Gate.h:
* Source/JavaScriptCore/runtime/JSCConfig.h:
* Source/JavaScriptCore/runtime/Options.cpp:
(JSC::Options::isAvailable):
(JSC::disableAllWasmOptions):
(JSC::Options::notifyOptionsChanged):
(JSC::Options::assertOptionsAreCoherent):
* Source/JavaScriptCore/runtime/OptionsList.h:
* Source/JavaScriptCore/runtime/SamplingProfiler.cpp:
(JSC::tierName):
* Source/JavaScriptCore/tools/JSDollarVM.cpp:
(JSC::JSC_DEFINE_HOST_FUNCTION):
* Source/JavaScriptCore/wasm/WasmBBQJIT.cpp:
(JSC::Wasm::BBQJITImpl::BBQJIT::addEndToUnreachable):
* Source/JavaScriptCore/wasm/WasmBBQPlan.cpp:
(JSC::Wasm::BBQPlan::work):
* Source/JavaScriptCore/wasm/WasmCallee.cpp:
(JSC::Wasm::Callee::runWithDowncast):
(JSC::Wasm::LLIntCallee::LLIntCallee): Deleted.
(JSC::Wasm::LLIntCallee::setEntrypoint): Deleted.
(JSC::Wasm::LLIntCallee::calleeSaveRegistersImpl): Deleted.
(JSC::Wasm::LLIntCallee::outOfLineJumpOffset): Deleted.
(JSC::Wasm::LLIntCallee::outOfLineJumpTarget): Deleted.
* Source/JavaScriptCore/wasm/WasmCallee.h:
* Source/JavaScriptCore/wasm/WasmCalleeGroup.cpp:
(JSC::Wasm::CalleeGroup::CalleeGroup):
(JSC::Wasm::CalleeGroup::releaseBBQCallee):
(JSC::Wasm::CalleeGroup::calleeIsReferenced const):
(JSC::Wasm::CalleeGroup::createFromLLInt): Deleted.
* Source/JavaScriptCore/wasm/WasmCalleeGroup.h:
* Source/JavaScriptCore/wasm/WasmCompilationMode.h:
(JSC::Wasm::isAnyInterpreter):
(JSC::Wasm::isAnyBBQ):
(JSC::Wasm::isAnyOMG):
(JSC::Wasm::isAnyWasmToJS):
* Source/JavaScriptCore/wasm/WasmEntryPlan.cpp:
(JSC::Wasm::EntryPlan::generateWasmToWasmStubs):
* Source/JavaScriptCore/wasm/WasmFaultSignalHandler.cpp:
(JSC::Wasm::trapHandler):
* Source/JavaScriptCore/wasm/WasmFunctionCodeBlockGenerator.cpp: Removed.
* Source/JavaScriptCore/wasm/WasmFunctionCodeBlockGenerator.h: Removed.
* Source/JavaScriptCore/wasm/WasmGeneratorTraits.h: Removed.
* Source/JavaScriptCore/wasm/WasmGlobal.h:
* Source/JavaScriptCore/wasm/WasmIPIntGenerator.cpp:
* Source/JavaScriptCore/wasm/WasmIPIntPlan.h:
* Source/JavaScriptCore/wasm/WasmIPIntSlowPaths.cpp:
(JSC::IPInt::wasm_log_crash):
(JSC::IPInt::slow_path_wasm_throw_exception):
(JSC::IPInt::slow_path_wasm_popcount):
(JSC::IPInt::slow_path_wasm_popcountll):
* Source/JavaScriptCore/wasm/WasmIPIntSlowPaths.h:
* Source/JavaScriptCore/wasm/WasmLLIntBuiltin.h: Removed.
* Source/JavaScriptCore/wasm/WasmLLIntGenerator.cpp: Removed.
* Source/JavaScriptCore/wasm/WasmLLIntGenerator.h: Removed.
* Source/JavaScriptCore/wasm/WasmLLIntPlan.cpp: Removed.
* Source/JavaScriptCore/wasm/WasmLLIntPlan.h: Removed.
* Source/JavaScriptCore/wasm/WasmLLIntTierUpCounter.cpp:
* Source/JavaScriptCore/wasm/WasmLLIntTierUpCounter.h:
* Source/JavaScriptCore/wasm/WasmModule.cpp:
(JSC::Wasm::Module::Module):
(JSC::Wasm::makeValidationCallback):
(JSC::Wasm::Module::validateSync):
(JSC::Wasm::Module::validateAsync):
(JSC::Wasm::Module::getOrCreateCalleeGroup):
* Source/JavaScriptCore/wasm/WasmModule.h:
* Source/JavaScriptCore/wasm/WasmModuleInformation.h:
* Source/JavaScriptCore/wasm/WasmOMGIRGenerator.cpp:
(JSC::Wasm::OMGIRGenerator::addCall):
* Source/JavaScriptCore/wasm/WasmOMGPlan.cpp:
(JSC::Wasm::OMGPlan::work):
* Source/JavaScriptCore/wasm/WasmOperations.cpp:
(JSC::Wasm::JSC_DEFINE_NOEXCEPT_JIT_OPERATION):
* Source/JavaScriptCore/wasm/WasmOperations.h:
* Source/JavaScriptCore/wasm/WasmOperationsInlines.h:
* Source/JavaScriptCore/wasm/WasmSlowPaths.cpp: Removed.
* Source/JavaScriptCore/wasm/WasmSlowPaths.h: Removed.
* Source/JavaScriptCore/wasm/WasmStreamingCompiler.cpp:
(JSC::Wasm::StreamingCompiler::didReceiveFunctionData):
(JSC::Wasm::StreamingCompiler::didFinishParsing):
(JSC::Wasm::StreamingCompiler::didComplete):
* Source/JavaScriptCore/wasm/WasmTypeDefinition.cpp:
(JSC::Wasm::TypeInformation::signatureForLLIntBuiltin): Deleted.
* Source/JavaScriptCore/wasm/WasmTypeDefinition.h:
* Source/JavaScriptCore/wasm/WasmWorklist.cpp:
* Source/JavaScriptCore/wasm/js/JSWebAssemblyInstance.cpp:
(JSC::JSWebAssemblyInstance::finalizeCreation):
* Source/JavaScriptCore/wasm/js/JSWebAssemblyStruct.cpp:
(JSC::JSWebAssemblyStruct::get const):
* Source/JavaScriptCore/wasm/js/WebAssemblyFunction.h:
* Tools/Scripts/run-jsc-stress-tests:

Canonical link: https://commits.webkit.org/298474@main
